catalog/models.py

```diff
@@ -1,3 +1,102 @@
 from django.db import models
+from django.urls import reverse
+import uuid
 
-# Create your models here.
+
+class Genre(models.Model):
+    """Model representing a book genre."""
+
+    name = models.CharField(
+        max_length=200, help_text="Enter a book genre (e.g. Science Fiction)"
+    )
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return self.name
+
+
+class Book(models.Model):
+    """Model representing a book (but not a specific copy of a book)."""
+
+    title = models.CharField(max_length=200)
+
+    # Foreign Key used because book can only have one author, but authors can have multiple books
+    # Author as a string rather than object because it hasn't been declared yet in the file
+    author = models.ForeignKey("Author", on_delete=models.SET_NULL, null=True)
```

Comment:
`on_delete` có những giá trị nào?

Reply:
*on_delete* có những giá trị như là:
+ CASCADE: Khi đối tượng được tham chiếu bị xóa, cũng xóa các đối tượng có tham chiếu đến nó (ví dụ trong bài này nếu để CASCADE thì nếu một tác giả bị xóa thì những quyển sách của ông cũng bị xóa theo)
+ PROTECT/RESTRICT: Cấm xóa đối tượng được tham chiếu. Ví dụ muốn xóa thì phải xóa tất cả các đối tượng tham chiếu đến nó theo cách thủ công
+ SET_NULL: Đặt tham chiếu thành NULL. Ví dụ: khi xóa một tác giả nào đó thì tất cả sách của ông vẫn còn nhưng mục tác giả sẽ hiển thị là NULL(vô danh)
+ SET_DEFAULT: Đặt giá trị mặc định.
+ SET (...): Đặt một giá trị cho trước. Đây không phải là một phần của tiêu chuẩn SQL và hoàn toàn do Django xử lý. Thay vì set default ta có thể set theo các giá trị mà ta đã định trước
+ DO_NOTHING: tham chiếu đến một đối tượng thực sự không tồn tại - no action. Tương đương với SQL: NO ACTION